### PR TITLE
Only invalidate efb copies when creating an efb copy

### PR DESCRIPTION
**Pull request by mimimi085181**
This should speed up a few games that use partial texture updates. More precisely those where the part that overlaps with the efb copy might be changed, and the rest is unchanged. Without this pr, Dolphin invalidates the texture, then recreates it basically identical. The part that overlaps with the efb copy does not matter, since it will be updated via partial texture updates.

Before partial texture updates, it was necessary to invalidate everything that overlaps an efb copy, at least when using efb to texture only. Now if efb copy and normal texture overlap, the plan is to update the texture with the efb copy.

This pr could only be considered for 5.0, when/if partial updates are applied for paletted textures as well.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3747)

### CODE REVIEW

**Comment by mimimi085181** [comment]
Ok, i really hope that difference on the fifo log from the simpsons game is from a paletted texture. If not, that's a clear reason against this pr. I guess now i really have to make that pr for partial updates for paletted textures.

File: Source/Core/VideoCommon/TextureCacheBase.cpp [hunk 1]
```diff
@@ -1196,15 +1199,17 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, unsigned int dstFo
 		copy_to_vram = false;
 	}
 
-	// Invalidate all textures that overlap the range of our efb copy.
-	// Unless our efb copy has a weird stride, then we want avoid invalidating textures which
-	// we might be able to do a partial texture update on.
+	// Invalidate all EFB copies that overlap the range of our efb copy. Any overlapping
```
Comment (lioncash) [inline Source/Core/VideoCommon/TextureCacheBase.cpp]: missing capitalization on `efb`

**Comment by delroth** [comment]
Optimizations are out of scope for 5.0.

**Comment by brujo5** [comment]
This commit speed up DBZ budokai tenkiachi 3 wii for android but cause this when the character is on the edge screen

screenshot_2016-03-25-22-20-14

**Comment by JMC47** [comment]
It needs Pull Request 3748 alongside it to prevent issues like those.

**Comment by phire** [comment]
There is no reason not to invalidate regular textures as well. You just want an exception for the resulting partial texture.

**Comment by brujo5** [comment]
Oh thanks for the advice @JMC47

**Comment by mimimi085181** [comment]
@phire: What do you recommend to do instead?
Imagine a game that updates a texture frame by frame in different places. Right now you would always only get the lastest update. So you would want to mark updated textures as updated and exclude them from this invalidation here? And possibly as well as on the garbage collection, if the hash is still valid, like efb copies are excluded there?

Well, for this pr to properly work, we need the partial updates for paletted textures(PR#3748) and i think we need to remove the restriction of using efb copies only for one update. I think i added that, because i saw the texture in new super mario bros was being loaded several times per frame. Would this be fixed by PR#3749, so the restriction can just be removed? I don't want that Dolphin is applying the same update 6 times per frame.

 And of course the partial updates code would need and update, if there's still a missing case.

**Comment by mimimi085181** [comment]
Closing this pr for now. It did not get the result i was hoping for. It also needs to be changed in a bigger way, in order to work properly..

**Pull request closed without merging**